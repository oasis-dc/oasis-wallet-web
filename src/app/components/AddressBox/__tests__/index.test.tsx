import copy from 'copy-to-clipboard'
import { screen } from '@testing-library/dom'
import { render } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

import { AddressBox, EditableAddressBox, EditableNameBox } from '../index'

jest.mock('copy-to-clipboard')

const testAddress = 'oasis1qqurxkgavtcjjytneumeclx59ds3avjaqg7ftqph'

const renderComponent = () => render(<AddressBox address={testAddress} />)

describe('<AddressBox />', () => {
  it('should render address properly', () => {
    const { container } = renderComponent()
    expect(container).toMatchSnapshot()
  })

  it('should be able to copy address to clipboard', async () => {
    renderComponent()
    await userEvent.click(screen.getByTestId('copy-address-button'))
    expect(copy).toHaveBeenCalledWith(testAddress)
  })

  it('should be able to show a notification', async () => {
    renderComponent()
    jest.mocked(copy).mockReturnValue(true) // Copy must return true so that the notification is actually displayed
    await userEvent.click(screen.getByTestId('copy-address-button'))
    expect(await screen.getByText('account.addressCopied')).toBeInTheDocument()
  })
})

const openEditModal = jest.fn()
const renderEditableNameBoxComponent = () =>
  render(<EditableNameBox address={testAddress} name="My Wallet" openEditModal={openEditModal} />)

describe('<EditableNameBox />', () => {
  it('should render name properly', () => {
    const { container } = renderEditableNameBoxComponent()
    expect(container).toMatchSnapshot()
  })

  it('should render edit icon', async () => {
    renderEditableNameBoxComponent()
    expect(screen.queryByTestId('editable-address-edit-button')).not.toBeInTheDocument()
    await userEvent.click(screen.getByTestId('editable-name-edit-button'))
    expect(openEditModal).toHaveBeenCalled()
  })

  it('should be able to copy address to clipboard', async () => {
    renderEditableNameBoxComponent()
    expect(screen.queryByTestId('copy-address-icon')).not.toBeInTheDocument()
    await userEvent.click(screen.getByTestId('copy-address-button'))
    expect(copy).toHaveBeenCalledWith(testAddress)
  })
})

const renderEditableAddressBoxComponent = () =>
  render(<EditableAddressBox openEditModal={openEditModal} address={testAddress} />)

describe('<EditableAddressBox />', () => {
  it('should render edit icon', async () => {
    renderEditableAddressBoxComponent()
    expect(screen.queryByTestId('editable-name-edit-button')).not.toBeInTheDocument()
    await userEvent.click(screen.getByTestId('editable-address-edit-button'))
    expect(openEditModal).toHaveBeenCalled()
  })

  it('should be able to copy address to clipboard', async () => {
    renderEditableAddressBoxComponent()
    await userEvent.click(screen.getByTestId('copy-address-button'))
    expect(copy).toHaveBeenCalledWith(testAddress)
  })
})
