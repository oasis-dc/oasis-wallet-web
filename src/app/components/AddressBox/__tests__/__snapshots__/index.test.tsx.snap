// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddressBox /> should render address properly 1`] = `
.c3 {
  display: inline-block;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 18px;
  height: 18px;
  fill: #666666;
  stroke: #666666;
}

.c3 g {
  fill: inherit;
  stroke: inherit;
}

.c3 *:not([stroke])[fill='none'] {
  stroke-width: 0;
}

.c3 *[stroke*='#'],
.c3 *[STROKE*='#'] {
  stroke: inherit;
  fill: none;
}

.c3 *[fill-rule],
.c3 *[FILL-RULE],
.c3 *[fill*='#'],
.c3 *[FILL*='#'] {
  fill: inherit;
  stroke: none;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding-right: 12px;
  border-radius: 5px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-right: 48px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: min(690px,100%);
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
}

.c5 {
  font-size: inherit;
  line-height: normal;
  font-weight: bold;
  word-break: break-word;
}

.c6 {
  font-size: inherit;
  line-height: normal;
}

.c2 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  color: inherit;
  outline: none;
  border: none;
  padding: 0;
  text-align: inherit;
  line-height: 0;
  padding: 12px;
}

.c2:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c2:focus > circle,
.c2:focus > ellipse,
.c2:focus > line,
.c2:focus > path,
.c2:focus > polygon,
.c2:focus > polyline,
.c2:focus > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c2:focus::-moz-focus-inner {
  border: 0;
}

.c2:focus:not(:focus-visible) {
  outline: none;
  box-shadow: none;
}

.c2:focus:not(:focus-visible) > circle,
.c2:focus:not(:focus-visible) > ellipse,
.c2:focus:not(:focus-visible) > line,
.c2:focus:not(:focus-visible) > path,
.c2:focus:not(:focus-visible) > polygon,
.c2:focus:not(:focus-visible) > polyline,
.c2:focus:not(:focus-visible) > rect {
  outline: none;
  box-shadow: none;
}

.c2:focus:not(:focus-visible)::-moz-focus-inner {
  border: 0;
}

@media only screen and (max-width:768px) {
  .c0 {
    padding-right: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    margin-right: 24px;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <button
        class="c2"
        data-testid="copy-address-button"
        type="button"
      >
        <svg
          aria-label="Copy"
          class="c3"
          viewBox="0 0 24 24"
        >
          <path
            d="M9 15h8-8zm0-4h10H9zm0-4h4-4zm7-6v6h6M6 5H2v18h16v-4m4 0H6V1h11l5 5v13z"
            fill="none"
            stroke="#000"
            stroke-width="2"
          />
        </svg>
      </button>
      <div
        class="c4"
      >
        <span
          class="c5"
          style="display: inline-flex; align-items: center; justify-content: space-between; flex: 1;"
        >
          <span
            class="c6"
            style="font-family: "Roboto mono", monospace; letter-spacing: 0;"
          >
            <span
              class="notranslate"
              translate="no"
            >
              oasis1 qqur xkga vtcj jytn eume clx5 9ds3 avja qg7f tqph
            </span>
          </span>
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`<EditableNameBox /> should render name properly 1`] = `
.c6 {
  display: inline-block;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 16px;
  height: 16px;
  fill: link;
  stroke: link;
}

.c6 g {
  fill: inherit;
  stroke: inherit;
}

.c6 *:not([stroke])[fill='none'] {
  stroke-width: 0;
}

.c6 *[stroke*='#'],
.c6 *[STROKE*='#'] {
  stroke: inherit;
  fill: none;
}

.c6 *[fill-rule],
.c6 *[FILL-RULE],
.c6 *[fill*='#'],
.c6 *[FILL*='#'] {
  fill: inherit;
  stroke: none;
}

.c10 {
  display: inline-block;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 18px;
  height: 18px;
  fill: #666666;
  stroke: #666666;
}

.c10 g {
  fill: inherit;
  stroke: inherit;
}

.c10 *:not([stroke])[fill='none'] {
  stroke-width: 0;
}

.c10 *[stroke*='#'],
.c10 *[STROKE*='#'] {
  stroke: inherit;
  fill: none;
}

.c10 *[fill-rule],
.c10 *[FILL-RULE],
.c10 *[fill*='#'],
.c10 *[FILL*='#'] {
  fill: inherit;
  stroke: none;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding-right: 12px;
  border-radius: 5px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-right: 48px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-bottom: solid 1px background-front-border;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: min(690px,100%);
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  padding-bottom: 6px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding-left: 12px;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c4 {
  font-size: inherit;
  line-height: normal;
  font-weight: bold;
  word-break: break-word;
}

.c5 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  color: inherit;
  outline: none;
  border: none;
  padding: 0;
  text-align: inherit;
  line-height: 0;
  padding: 12px;
}

.c5:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c5:focus > circle,
.c5:focus > ellipse,
.c5:focus > line,
.c5:focus > path,
.c5:focus > polygon,
.c5:focus > polyline,
.c5:focus > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c5:focus::-moz-focus-inner {
  border: 0;
}

.c5:focus:not(:focus-visible) {
  outline: none;
  box-shadow: none;
}

.c5:focus:not(:focus-visible) > circle,
.c5:focus:not(:focus-visible) > ellipse,
.c5:focus:not(:focus-visible) > line,
.c5:focus:not(:focus-visible) > path,
.c5:focus:not(:focus-visible) > polygon,
.c5:focus:not(:focus-visible) > polyline,
.c5:focus:not(:focus-visible) > rect {
  outline: none;
  box-shadow: none;
}

.c5:focus:not(:focus-visible)::-moz-focus-inner {
  border: 0;
}

.c7 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  color: inherit;
  outline: none;
  border: none;
  padding: 0;
  text-align: inherit;
}

.c7:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c7:focus > circle,
.c7:focus > ellipse,
.c7:focus > line,
.c7:focus > path,
.c7:focus > polygon,
.c7:focus > polyline,
.c7:focus > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c7:focus::-moz-focus-inner {
  border: 0;
}

.c7:focus:not(:focus-visible) {
  outline: none;
  box-shadow: none;
}

.c7:focus:not(:focus-visible) > circle,
.c7:focus:not(:focus-visible) > ellipse,
.c7:focus:not(:focus-visible) > line,
.c7:focus:not(:focus-visible) > path,
.c7:focus:not(:focus-visible) > polygon,
.c7:focus:not(:focus-visible) > polyline,
.c7:focus:not(:focus-visible) > rect {
  outline: none;
  box-shadow: none;
}

.c7:focus:not(:focus-visible)::-moz-focus-inner {
  border: 0;
}

.c9 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 12px;
  font-weight: 700;
  padding-top: 6px;
  padding-bottom: 6px;
  background-color: background-custom-2;
  border-radius: 18px;
  color: text-custom;
  padding-right: 24px;
  padding-left: 24px;
  border: solid 1px #7D4CDB;
}

@media only screen and (max-width:768px) {
  .c0 {
    padding-right: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    margin-right: 24px;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    border-bottom: solid 1px background-front-border;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    padding-bottom: 3px;
  }
}

@media only screen and (max-width:768px) {
  .c3 {
    padding-left: 6px;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <span
            class="c4"
            style="display: inline-flex; align-items: center; justify-content: space-between; flex: 1;"
          >
            My Wallet
          </span>
          <button
            class="c5"
            type="button"
          >
            <svg
              aria-label="Edit"
              class="c6"
              data-testid="editable-name-edit-button"
              viewBox="0 0 24 24"
            >
              <path
                d="m14 4 6 6-6-6zm8.294 1.294c.39.39.387 1.025-.008 1.42L9 20l-7 2 2-7L17.286 1.714a1 1 0 0 1 1.42-.008l3.588 3.588zM3 19l2 2m2-4 8-8"
                fill="none"
                stroke="#000"
                stroke-width="2"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
    <button
      class="c7"
      data-testid="copy-address-button"
      type="button"
    >
      <div
        class="c8 c9"
      >
        oasis1qq...7ftqph
        <svg
          aria-label="Copy"
          class="c10"
          viewBox="0 0 24 24"
        >
          <path
            d="M9 15h8-8zm0-4h10H9zm0-4h4-4zm7-6v6h6M6 5H2v18h16v-4m4 0H6V1h11l5 5v13z"
            fill="none"
            stroke="#000"
            stroke-width="2"
          />
        </svg>
      </div>
    </button>
  </div>
</div>
`;
